Assets/Shopify/UIToolkit/Shop Controllers/Editor/SingleProductShopControllerEditor.cs

```diff
@@ -32,7 +32,11 @@ public class SingleProductShopControllerEditor : ShopControllerBaseEditor, ISing
 
         private ProductPicker _productPicker {
             get {
-                _cachedPicker = _cachedPicker ?? new ProductPicker(Client);
+                if (_cachedPicker == null) {
+                    _cachedPicker = new ProductPicker(Client);
+                    _cachedPicker.OnProductListUpdated += ProductListDidUpdate;
```

Comment:
Can you just add `Repaint` directly, without an extra method in between? Or maybe just a lambda?

Reply:
Sure, I thought it was a bit clearer from like a high level that things happen when the product list updates in that callback method or whatever?